index.d.ts

```diff
@@ -323,6 +323,25 @@ interface ConfigParams {
    */
   getLoginState?: (req: OpenidRequest, options: LoginOptions) => object;
 
+  /**
+   * Function for custom callback handling after receiving tokens and before redirecting
+   * This can be used for handling token storage, making userinfo calls, claim validation, etc.
+   *
+   * ```js
+   * app.use(auth({
+   *   ...
+   *   afterCallback(req, res, session, decodedState) {
+   *     ...
+   *     return {
+   *       ...session,
+   *       ...additionalUserClaims
+   *     };
+   *   }
+   * }))
+   * ``
+   */
+  afterCallback?: (req: OpenidRequest, res: OpenidResponse, session: object, decodedState: object) => object;
```

Comment:
Let's call it a `tokenSet` and type it as a `TokenSet`

Reply:
We do not expose any internals of the `openid-client` library, i suggest we don't start doing so here.

Nevertheless the types `session: object, decodedState: object` don't allow developers to work with the arguments without explicitly casting them to something they know is on them. I suggest typing these with what we know is there + having a `[propName: string]: any` indexable property.

Furthermore the function type should allow the result to be a promise. 

```
type AsyncOrSync<T> = PromiseLike<T> | T

afterCallback?: (...) => AsyncOrSync<resultInterface>;
```